src/Strategy.sol

```diff
@@ -294,9 +311,20 @@ contract Strategy is BaseStrategy {
     // transfers all tokens to new strategy
     function prepareMigration(address _newStrategy) internal override {
         // want is transferred by the base contract's migrate function
-        sanTokenGauge.withdraw(balanceOfStakedSanToken());
 
-        IERC20(sanToken).safeTransfer(_newStrategy, balanceOfSanToken());
+        uint256 _angleBalance = balanceOfAngleToken();
+        if (_angleBalance > 0) {
+            IERC20(angleToken).safeTransfer(_newStrategy, _angleBalance);
+        }
+
+        uint256 _stakedBalance = balanceOfStakedSanToken();
+        if (_stakedBalance > 0) {
+            strategyProxy.withdraw(address(sanTokenGauge), address(sanToken), _stakedBalance);
```

Comment:
why not `withdrawAll`?

Reply:
I forgot about its existence, should be same result but replacing as it's exactly its purpose!